stream/stream.go

```diff
@@ -42,6 +44,27 @@ func (k key) DecryptPipe(r io.Reader, w io.Writer, extra ...[]byte) error {
 	return nil
 }
 
+var buffers = sync.Pool{
+	New: func() interface{} {
+		return new(bytes.Buffer)
+	},
+}
+
+func GetBuffer() (*bytes.Buffer, error) {
```

Comment:
The functions `GetBuffer` and `PutBuffer` should be unexported, or the clients of the package could use them and insert their own buffers.